hbase-protocol-shaded/src/main/protobuf/server/master/MasterProcedure.proto

```diff
@@ -102,6 +102,19 @@ message TruncateTableStateData {
   repeated RegionInfo region_info = 5;
 }
 
+enum TruncateRegionState {
+  TRUNCATE_REGION_PRE_OPERATION = 1;
+  TRUNCATE_REGION_MAKE_OFFLINE = 2;
+  TRUNCATE_REGION_REMOVE = 3;
+  TRUNCATE_REGION_MAKE_ONLINE = 4;
+  TRUNCATE_REGION_POST_OPERATION = 5;
+}
+
+message TruncateRegionStateData {
+  required UserInformation user_info = 1;
+  required RegionInfo region = 2;
```

Comment:
nit: could we rename this to region_info, same is being used at most places

Reply:
I have removed the TruncateRegionStateData. As you suggested in earlier comment, we could reuse the filed from the parent and rely on serialisation/deserialisation implemented by superClass.